src/path.h

```diff
@@ -41,17 +41,19 @@ int path_get_config_is_remote();
 class env_stack_t;
 void path_emit_config_directory_messages(env_stack_t &vars);
 
-/// Finds the full path of an executable.
+/// Finds the path of an executable.
 ///
 /// Args:
 /// cmd - The name of the executable.
-/// output_or_NULL - If non-NULL, store the full path.
+/// output_or_NULL - If non-NULL, store the path.
 /// vars - The environment variables to use
+/// absolute - Should returned path be absolute (if it is set to false, returned path may be either
+///            absolute or relative depending on the content of cmd argument)
 ///
 /// Returns:
 /// false if the command can not be found else true. The result
 /// should be freed with free().
-bool path_get_path(const wcstring &cmd, wcstring *out_path, const environment_t &vars);
+bool path_get_path(const wcstring &cmd, wcstring *out_path, const environment_t &vars, bool absolute);
```

Comment:
This is a case where I'd use a default argument - `bool absolute = false` and just leave it out in the spots you don't need it.

Then, where you do, you can comment it as

```fish
path_get_path(str, path, vars, /* absolute */ true)
```

Alternatively we could require those parts to do the `wrealpath` themselves. If it's only used in one or two places (and I count *one* where I'd agree that we need it) that might be preferable.

Reply:
Yeah I think we only want this in one place, so the caller should call `wrealpath`.

Reply:
Agreed, this makes more sense. I just removed information from `path_get_path` that it returns full path and call `wrealpath` in `parse_cmd_string`.